### ISSUE TITLE
ProvisionedMeshNode Parcelable implementation broken

### ISSUE THREAD

**Issue by HaukeRa** [NordicSemiconductor/Android-nRF-Mesh-Library#132]
---

**Describe the bug**
Generally Parcelable objects are expected to contain the same information before and after parceling them and retreiving them from a parcel.

Two issues:
1. ProvisionedMeshNode(Parcel in):
`110:  isProvisioned = in.readByte() != 1;
 111: isConfigured = in.readByte() != 1;
 137: blackListed = in.readByte() != 1`
inverts the state written to the parcel (written as true --> 1, false -->0. read as 1-->false, 0-->true)
2. Deprecated fields not included in parcelable (not so bad, as they are deprecated anyways)
Example: NetworkKey in BaseProvisionedMeshNode is always null after reading ProvisionedMeshNode from parcel

**Expected behavior**
Contents of objects equal before and after parceling/unparceling

**Comment by HaukeRa** [comment]
Another error:

* blackListed is written with Parcel.writeValue() but read with Parcel.readByte()

This results in an error in SOME cases. I spent all day trying to write a test for this, just now I found a way to test this.

While it works in most cases, sometimes it results in a hard to debug crash. Internally Parcel.writeValue writes two Integers, one as a flag for what type has been written and one for the value itself. Parcel.writeByte only writes the value itself. So using not the corresponding method results in a partially read Parcel, so subsequent reads might fail with weird errors like: `java.lang.RuntimeException: Parcel android.os.Parcel@3ed21c6: Unmarshalling unknown type code 7209057 at offset 2012.`
